ConditionFormats/Serialization: Update eos portable archive to work with boost 1.69.0+

gartung opened the pull request:
#### PR description:
Update eos portable archive to work with boost 1.69.0+
Taken from https://github.com/daldegam/eos-portable-archive/pull/2
This replaces include path to fpclassify.hpp and endian.hpp for changes in 1.69.00 and changes the functions called based on boost version or boost header settings.

#### PR validation:

ConditionFormats tests?

NB. Updating to boost 1.69.0 also changes the signals interface which breaks the build because libboost_signals is no longer built.

gartung committed:
Update eos portable archive to work with boost 1.69.0+

Taken from https://github.com/daldegam/eos-portable-archive/pull/2

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13673

- This PR adds an extra 20KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13673/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13673/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

gartung committed:
code format changes

cmsbuild commented:
The code-checks are being triggered in jenkins.

smuzaffar commented:
thanks @gartung , did not know that it was taken from https://github.com/daldegam/eos-portable-archive . Are there any cms changes on top of daldegam/eos-portable-archive ? How about moving it to external?

FYI @ggovi

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13674

- This PR adds an extra 24KB to repository

cmsbuild commented:
A new Pull Request was created by  @gartung (Patrick Gartung) for master.

It involves the following packages:

CondFormats/Serialization


@ggovi, @cmsbuild can you please review it and eventually sign? Thanks.
@mmusich, @seemasharmafnal this is something you requested to watch as well.
@davidlange6, @silviodonato, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

gartung commented:
I don't know that this was taken from there. I searched github for the error message I am getting trying to build with boost 1.70.0 and found that pull request. 

A search of github shows many copies of eos header names.

There are build errors in 
CondFormats/Serialization/src/templateInstantiations.cc
that I am trying to understand.

gartung commented:
There are two repos on github for eos-portable-archive that have pr's with the update

https://github.com/search?q=eos-portable-archive&type=Repositories

https://github.com/duncanka/eos-portable-archive/pull/2

The second repo at least links to the original project page:

https://archive.codeplex.com/?p=epa

which has a link to the original archive.

My web browser generated a warning about the website though.

gartung commented:
The errors I am seeing are for duplicate instantiations which seems to be related to this change
```
commit 8369ace830dbadbaef3969d4d1f60b466bd560ba
Author: Andreas Pfeiffer <andreas.pfeiffer@cern.ch>
Date:   Fri Oct 3 09:43:36 2014 +0200

    moved template instantiation out of eos files into CondFormats/Serialization lib, added usage of CondFormats/Serialization to BuildFiles who need the lib

```

gartung committed:
Run unix2dos for better comparison

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13676

- This PR adds an extra 32KB to repository

cmsbuild commented:
Pull request #28893 was updated. @ggovi, @cmsbuild can you please check and sign again.

gartung committed:
Add changes made after eos added to ConditionFormats/Serialization

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13677

- This PR adds an extra 32KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13677/code-format.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13677/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

gartung committed:
Code formatting

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13678

- This PR adds an extra 32KB to repository

cmsbuild commented:
Pull request #28893 was updated. @ggovi, @cmsbuild can you please check and sign again.

gartung commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

gartung commented:
@Dr15Jones is boost::signals used anywhere is CMSSW anymore. If not we can update the boost toolfiles to remove references to libboost_signal which does not exist in boost 1.69.0+

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmsdist#5526

cmsbuild commented:
+1
Tested at: 9c4ade4bae78a28430d4c38dd298ee6b27c6c6e2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c37db9/4556/summary.html
CMSSW: CMSSW_11_1_X_2020-02-07-1100
SCRAM_ARCH: slc7_amd64_gcc820

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #28753 @cericeci: Fixed sign in pat::Muon::dB() to match one obtained from the best track definition 
 - #28874 @kpedro88: Make FastTimerService easier to use
 - #28872 @rovere: HGCAL Validation technical changes
 - #28868 @bsunanda: Run3 hcx249 Test and verify HcalParameter correctly extracted in the dd4hep scenario
 - #28873 @intrepid42: Basic Rivet tests

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c37db9/4556/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c37db9/4556/git-merge-result

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c37db9/4556/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 12 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2698043
 - DQMHistoTests: Total failures: 4926
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2692771
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

VinInn commented:
why all this "BOOST_VERSION"?
One of the advantage of CMSSW is that each version has a well defined set of external.
This is done ALSO to avoid to have to support an infinite number of combinations of "external".

gartung commented:
I don't know why it was put into CMSSW in the first place. The source code is directly from the the original.

smuzaffar committed:
fixes needd for boost 1.72

cmsbuild commented:
The code-checks are being triggered in jenkins.

smuzaffar commented:
please test with cms-sw/cmsdist#5526

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13686

- This PR adds an extra 32KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmsdist#5526

cmsbuild commented:
Pull request #28893 was updated. @benkrikler, @christopheralanwest, @tocheng, @cmsbuild, @rekovic, @franzoni, @tlampen, @ggovi, @pohsun can you please check and sign again.

cmsbuild commented:
-1

Tested at: ddbef72af21c8d5e79985ba9278b098c9b2ceae5

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw/slc7_amd64_gcc820/cms/coral/CORAL_2_3_21-dbd688/src/LCG/ConnectionService/src/ConnectionHandle.cpp 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw/slc7_amd64_gcc820/cms/coral/CORAL_2_3_21-dbd688/include/LCG -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/boost/1.67.0-pafccj/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/libuuid/2.34/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/xerces-c/3.1.3-pafccj/include -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++1z -ftree-vectorize -Wstrict-overflow -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -fdebug-prefix-map=/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir=/opt/cmssw -fdebug-prefix-map=/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw=/opt/cmssw -g   -fPIC    -MMD -MF tmp/slc7_amd64_gcc820/src/LCG/ConnectionService/src/lcg_ConnectionService/ConnectionHandle.cpp.d /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw/slc7_amd64_gcc820/cms/coral/CORAL_2_3_21-dbd688/src/LCG/ConnectionService/src/ConnectionHandle.cpp -o tmp/slc7_amd64_gcc820/src/LCG/ConnectionService/src/lcg_ConnectionService/ConnectionHandle.cpp.o
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw/slc7_amd64_gcc820/cms/coral/CORAL_2_3_21-dbd688/src/LCG/ConnectionService/src/SessionProxy.cpp 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/gcc/8.2.0-pafccj/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw/slc7_amd64_gcc820/cms/coral/CORAL_2_3_21-dbd688/include/LCG -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/boost/1.67.0-pafccj/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/libuuid/2.34/include -I/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/slc7_amd64_gcc820/external/xerces-c/3.1.3-pafccj/include -O2 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++1z -ftree-vectorize -Wstrict-overflow -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -msse3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wunused -Wparentheses -Wno-deprecated -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -fdebug-prefix-map=/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir=/opt/cmssw -fdebug-prefix-map=/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw=/opt/cmssw -g   -fPIC    -MMD -MF tmp/slc7_amd64_gcc820/src/LCG/ConnectionService/src/lcg_ConnectionService/SessionProxy.cpp.d /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/BUILDROOT/c95ee7b21689f520b9e247878b4166c9/opt/cmssw/slc7_amd64_gcc820/cms/coral/CORAL_2_3_21-dbd688/src/LCG/ConnectionService/src/SessionProxy.cpp -o tmp/slc7_amd64_gcc820/src/LCG/ConnectionService/src/lcg_ConnectionService/SessionProxy.cpp.o
+ '[' '' == set ']'
error: Bad exit status from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/rpm-tmp.9H0N9r (%build)


RPM build errors:
Bad exit status from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/tmp/rpm-tmp.9H0N9r (%build)

</pre>

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-dbd688/4563/summary.html

smuzaffar commented:
please test with cms-sw/cmsdist#5526,cms-sw/cmsdist#5387

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmsdist#5526,cms-sw/cmsdist#5387

cmsbuild commented:
-1

Tested at: ddbef72af21c8d5e79985ba9278b098c9b2ceae5

CMSSW: CMSSW_11_1_X_2020-02-09-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5ab8a/4564/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test test_MuonGeometryDBConverter had ERRORS
---> test testPayloadProxy had ERRORS
---> test testConditionDatabase_0 had ERRORS
---> test testSerializationAlignment had ERRORS
---> test testSerializationBTauObjects had ERRORS
---> test testSerializationCalibration had ERRORS
---> test testSerializationDTObjects had ERRORS
---> test testSerializationLuminosity had ERRORS
---> test testSerializationRunInfo had ERRORS
---> test condTestRegression had ERRORS
---> test GeometryDTGeometryBuilderTestDriver had ERRORS

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5ab8a/4564/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2698043
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2697696
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

smuzaffar committed:
Update portable_oarchive.hpp

smuzaffar commented:
please test with cms-sw/cmsdist#5526,cms-sw/cmsdist#5387

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-28893/13706

- This PR adds an extra 48KB to repository

cmsbuild commented:
The tests are being triggered in jenkins.
Tested with other pull request(s) cms-sw/cmsdist#5526,cms-sw/cmsdist#5387

cmsbuild commented:
Pull request #28893 was updated. @benkrikler, @christopheralanwest, @tocheng, @cmsbuild, @rekovic, @franzoni, @tlampen, @ggovi, @pohsun can you please check and sign again.

cmsbuild commented:
-1

Tested at: 5cfa0b5b9845c789ef045751f2f0920eda02cbe7

CMSSW: CMSSW_11_1_X_2020-02-10-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5ab8a/4587/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test GeometryDTGeometryBuilderTestDriver had ERRORS

cmsbuild commented:
Comparison job queued.

smuzaffar commented:
please test
lets re-run the tests using old boost i.e 1.67. If every things goes fine then we can merge it for nromal IBs and start testing ne boost in DEVEL IBs.

By the way, unit test failure has nothing to do with this PR, test is already failing for IBs.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-d5ab8a/4587/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2694005
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2693658
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

cmsbuild commented:
+1
Tested at: 5cfa0b5b9845c789ef045751f2f0920eda02cbe7
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c37db9/4604/summary.html
CMSSW: CMSSW_11_1_X_2020-02-10-2300
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c37db9/4604/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 34
 - DQMHistoTests: Total histograms compared: 2694005
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2693658
 - DQMHistoTests: Total skipped: 346
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 33 files compared)
 - Checked 147 log files, 16 edm output root files, 34 DQM output files

smuzaffar commented:
@ggovi changes in CondFormats/Serialization looks good and they only effect boost version >= 1.69 which we do not have yet in cmssw. So this PR is safe to go in with boost 1.67 (which is what we have in cmssw). 

We have tested these changes with boost version 1.72 ( https://github.com/cms-sw/cmssw/pull/28893#issuecomment-584691286 ) and it does not break any thing. I would suggest to merge this in CMSSW and start  testing boost 1.72 in DEVEL IB.

ggovi commented:
@smuzaffar 
sounds good, thanks!

ggovi commented:
+1

smuzaffar commented on CalibCalorimetry/HcalAlgos/src/HcalDbHardcode.cc:
```diff
@@ -7,6 +7,7 @@
 #include <sstream>
 #include <iostream>
 #include <memory>
+#include <cassert>
```
@franzoni , @pohsun , @tlampen , @tocheng , @christopheralanwest can you please sign this change for alca?

smuzaffar commented on L1Trigger/L1TMuonBarrel/interface/L1TMuonBarrelParamsAllPublic.h:
```diff
@@ -1,6 +1,8 @@
 #ifndef L1TMuonBarrelParamsAllPublic_h
 #define L1TMuonBarrelParamsAllPublic_h
 
+#include <cassert>
```
@benkrikler,  @rekovic can you please sign this ( and https://github.com/cms-sw/cmssw/pull/28893/files#diff-d0dc19617f53a94d299fd1f591d10c6fR10 ) for `l1` so that we can proceed with boost 1.72 testing in DEVEL IBs?

christopheralanwest commented:
+1

smuzaffar commented:
@silviodonato , can we get ths in now? This will help testing boost 1.72 in DEVEL IBs.

smuzaffar commented:
@silviodonato , can we get this in? only L1 signatures are missing and changes for L1 are trivial (included just `assert` header)

rekovic commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @davidlange6, @silviodonato, @fabiocos (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
